deepchecks/utils/distribution/prediction_drift.py

```diff
@@ -0,0 +1,113 @@
+# ----------------------------------------------------------------------------
+# Copyright (C) 2021-2023 Deepchecks (https://www.deepchecks.com)
+#
+# This file is part of Deepchecks.
+# Deepchecks is distributed under the terms of the GNU Affero General
+# Public License (version 3 or later).
+# You should have received a copy of the GNU Affero General Public License
+# along with Deepchecks.  If not, see <http://www.gnu.org/licenses/>.
+# ----------------------------------------------------------------------------
+#
+"""Module contains the Abstract cass for Prediction Drift checks."""
```

Comment:
maybe put it under core - abstract checks? (also move abstract weak segments there)